iron-form.html

```diff
@@ -11,503 +11,459 @@
           }
+          this.$.helper.submit();
+          this.fire('iron-form-submit');
         } else {
-          addSerializedElement(el.name, el.value);
+          this._makeAjaxRequest(json);
         }
-      }
+      },
 
-      return json;
-    },
+      /**
+       * Resets the form to the default values.
+       */
+      reset: function(event) {
+        // We are not using this form for submission, so always cancel its event.
+        if (event)
+          event.preventDefault();
+
+        // If you've called this before distribution happened, bail out.
+        if (!this._form)
+          return;
+
+        // Load the initial values.
+        var nodes = this._getSubmittableElements();
+        for (var i = 0; i < nodes.length; i++) {
+          var node = nodes[i];
+          if (this._defaults.has(node)) {
+            var defaults = this._defaults.get(node);
+            if (node.value != null) {
+              node.value = defaults.value;
+            }
+            if (node.checked != null) {
+              node.checked = defaults.checked;
+            }
+          }
+        }
+      },
 
-    _handleFormResponse: function (event) {
-      this.fire('iron-form-response', event.detail);
-    },
+      /**
+       * Serializes the form as will be used in submission. Note that `serialize`
+       * is a Polymer reserved keyword, so calling `someIronForm`.serialize()`
+       * will give you unexpected results.
+       * @return {Object} An object containing name-value pairs for elements that
+       *                  would be submitted.
+       */
+      serializeForm: function() {
+        // Only elements that have a `name` and are not disabled are submittable.
+        var elements = this._getSubmittableElements();
+        var json = {};
+        for (var i = 0; i < elements.length; i++) {
+          var values = this._serializeElementValues(elements[i]);
+          for (var v = 0; v < values.length; v++) {
+            this._addSerializedElement(json, elements[i].name, values[v]);
+          }
+        }
+        return json;
+      },
 
-    _handleFormError: function (event) {
-      this.fire('iron-form-error', event.detail);
-    },
+      _handleFormResponse: function (event) {
+        this.fire('iron-form-response', event.detail);
+      },
 
-    _registerElement: function(e) {
-      // Get the actual element that fired the event
-      var element = Polymer.dom(e).rootTarget;
+      _handleFormError: function (event) {
+        this.fire('iron-form-error', event.detail);
+      },
 
-      element._parentForm = this;
-      this._customElements.push(element);
+      _makeAjaxRequest: function(json) {
+        // Initialize the iron-ajax element if we haven't already.
+        if (!this.request) {
+          this.request = document.createElement('iron-ajax');
+          this.request.addEventListener('response', this._handleFormResponse.bind(this));
```

Comment:
i think it can emit them if it fires them manually, but not the `-changed` ones Polymer should set up...right? So this should be ok?